### PR TITLE
Feature/us1446  create post for job

### PR DESCRIPTION
**Pull request by danrye**

### CODE REVIEW

File: Gateway/CorkboardAPI/CorkboardAPI.Tests/Controllers/PostsControllerTest.cs [hunk 1]
```diff
@@ -14,16 +14,16 @@ public class PostsControllerTest
         [TestCase]
         public void GetCorkboardPostsTest()
         {
-            // Arrange
-            var controller = new PostsController();
-            controller.Request = new HttpRequestMessage();
-            controller.Configuration = new HttpConfiguration();
+            //// Arrange
```
Comment (danrye) [inline Gateway/CorkboardAPI/CorkboardAPI.Tests/Controllers/PostsControllerTest.cs]: @jpc11623 should we remove this test, or fix it... why's this commented out?

File: Gateway/CorkboardAPI/CorkboardAPI/App_Start/WebApiConfig.cs [hunk 2]
```diff
@@ -21,18 +21,42 @@ public static void Register(HttpConfiguration config)
             // Web API routes
             config.MapHttpAttributeRoutes();
 
+            //config.Routes.MapHttpRoute(
```
Comment (danrye) [inline Gateway/CorkboardAPI/CorkboardAPI/App_Start/WebApiConfig.cs]: @jpc11623 clean up these comments

File: website/app/corkboard/directives/corkboard.card.directive.js [hunk 3]
```diff
@@ -8,7 +8,7 @@
   function CorkboardCard(){
     return {
       restrict: "EA",
-      templateUrl : "corkboard/corkboard.card.html",
+      templateUrl : "app/corkboard/templates/corkboard.card.html",
```
Comment (danrye) [inline website/app/corkboard/directives/corkboard.card.directive.js]: @danrye Need to change this to /templates/corkboard.card.html

File: Gateway/CorkboardAPI/CorkboardAPI/Services/PostService.cs [hunk 4]
```diff
@@ -85,6 +75,70 @@ public IHttpActionResult GetAllPosts()
             }
         }
 
+        public CorkboardPost SaveNeedPost(CorkboardNeedPost postItem, dynamic auth)
+        {
+            // connect to the mongodb server
```
Comment (IngageBrewerM) [inline Gateway/CorkboardAPI/CorkboardAPI/Services/PostService.cs]: Should catch mongo exceptions here and return a custom exception to the calling controller method which can be serialized correctly and handled on the client.

File: Gateway/CorkboardAPI/CorkboardAPI/Services/PostService.cs [hunk 5]
```diff
@@ -85,6 +75,70 @@ public IHttpActionResult GetAllPosts()
...
+
+            // mongo will assign an id to the doc being inserted, so we pull that and apply to the post
+            var id = needPostBsonDoc.GetValue("_id").AsObjectId.ToString();
+
+            return GetPostById(id, corkboardPosts);
+        }
+
+        public CorkboardPost SaveJobPost(CorkboardJobPost postItem, dynamic auth)
+        {
+            // connect to the mongodb server
```
Comment (IngageBrewerM) [inline Gateway/CorkboardAPI/CorkboardAPI/Services/PostService.cs]: See comment above

File: Gateway/CorkboardAPI/CorkboardAPI/Services/PostService.cs (hunk 4 above)
Comment (jpc11623) [inline Gateway/CorkboardAPI/CorkboardAPI/Services/PostService.cs]: We're planning a Tech Task for the next sprint to do some Mongo configuration and tuning. We will address more specific errors from Mongo in the API at that point.

File: Gateway/CorkboardAPI/CorkboardAPI.Tests/Controllers/PostsControllerTest.cs (hunk 1 above)
Comment (jpc11623) [inline Gateway/CorkboardAPI/CorkboardAPI.Tests/Controllers/PostsControllerTest.cs]: We're holding off on testing for the moment.

**Pull request merged**